From: "Rafael J. Wysocki" <rafael.j.wysocki@intel.com>
Date: Tue, 18 Nov 2014 01:43:42 +0100
Subject: [PATCH 66/77] PM: Kconfig: Set PM_RUNTIME if PM_SLEEP is selected
Patch-mainline: v3.19-rc1
Git-commit: b2b49ccbdd547135c69371ed066cffa44912060a
References: bsc#922836 Fate#318523

The number of and dependencies between high-level power management
Kconfig options make life much harder than necessary.  Several
conbinations of them have to be tested and supported, even though
some of those combinations are very rarely used in practice (if
they are used in practice at all).  Moreover, the fact that we
have separate independent Kconfig options for runtime PM and
system suspend is a serious obstacle for integration between
the two frameworks.

To overcome these difficulties, always select PM_RUNTIME if PM_SLEEP
is set.  Among other things, this will allow system suspend callbacks
provided by bus types and device drivers to rely on the runtime PM
framework regardless of the kernel configuration.

Enthusiastically-acked-by: Kevin Hilman <khilman@linaro.org>
Tested-by: Geert Uytterhoeven <geert+renesas@glider.be>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 kernel/power/Kconfig |    1 +
 1 file changed, 1 insertion(+)
--- a/kernel/power/Kconfig
+++ b/kernel/power/Kconfig
@@ -94,6 +94,7 @@ config PM_STD_PARTITION
 config PM_SLEEP
 	def_bool y
 	depends on SUSPEND || HIBERNATE_CALLBACKS
+	select PM_RUNTIME
 
 config PM_SLEEP_SMP
 	def_bool y
